tests/test_arrow_dataset.py

```diff
@@ -702,8 +702,12 @@ def test_concatenate_with_no_task_templates(self, in_memory):
                         self.assertEqual(dset_concat.info.task_templates, None)
 
     def test_concatenate_with_equal_task_templates(self, in_memory):
-        task_template = TextClassification(text_column="text", label_column="labels", labels=["pos", "neg"])
-        info = DatasetInfo(task_templates=task_template)
+        labels = ["neg", "pos"]
+        task_template = TextClassification(text_column="text", label_column="labels", labels=labels)
+        info = DatasetInfo(
+            features=Features({"text": Value("string"), "labels": ClassLabel(names=labels)}),
+            task_templates=task_template,
+        )
```

Comment:
Maybe add a test to make sure that the DatasetInfo post init does pass the labels to the task templates, so that we can instantiate the task_template without labels as in the demo ?

Reply:
good idea! i've now set up the unit test to cover the following cases:

1. `DatasetInfo` defined with a `TextClassification` template without `labels`
2. `DatasetInfo` defined with a `TextClassification` template with `labels` (this one might be overkill, but best to be safe)
3. Passing a `TextClassification` template with labels directly to `prepare_for_task` (the no labels case is covered in `test_task_with_incompatible_templates`)